packages/rum-core/test/performance-monitoring/transaction-service.spec.js

```diff
@@ -358,4 +377,43 @@ describe('TransactionService', function() {
     span1.end()
     tr1.detectFinish()
   })
+
+  it('should create unmanaged transactions by default', () => {
+    expect(transactionService.currentTransaction).toBeUndefined()
+    const tr1 = transactionService.startTransaction('test-name', 'test-type')
+    expect(tr1.name).toBe('test-name')
+    expect(transactionService.currentTransaction).toBeUndefined()
+    spyOn(tr1, 'removeTask')
+    spyOn(tr1, 'startSpan')
+    transactionService.removeTask('testId')
+    expect(tr1.removeTask).not.toHaveBeenCalled()
+    transactionService.startSpan('test-name', 'test-type')
+    expect(tr1.startSpan).not.toHaveBeenCalled()
+    expect(tr1.spans.length).toBe(0)
+  })
+
+  it('should create managed transactions if the managed option is provided', () => {
+    expect(transactionService.currentTransaction).toBeUndefined()
+    const tr1 = transactionService.startTransaction('test-name', 'test-type', {
+      managed: true
+    })
+    expect(tr1.name).toBe('test-name')
+    expect(transactionService.currentTransaction).toBe(tr1)
+
+    const span = transactionService.startSpan(
+      'test-span-name',
+      'test-span-type'
+    )
+    span.end()
+    expect(tr1.spans[0]).toBe(span)
+    spyOn(tr1, 'addTask').and.callThrough()
+    spyOn(tr1, 'removeTask').and.callThrough()
+    spyOn(tr1, 'end')
+    transactionService.addTask('taskId')
+    expect(tr1.addTask).toHaveBeenCalledWith('taskId')
+
+    transactionService.removeTask('taskId')
+    expect(tr1.removeTask).toHaveBeenCalledWith('taskId')
+    expect(tr1.end).toHaveBeenCalled()
+  })
```

Comment:
Can we add another test where we add two transactions 1. managed and 2. unmanaged and call start span accordingly and check if they are getting added to the right thing. It provides some guarantees that both the transactions can coexist on the page.